hydradx/model/amm/trade_strategies.py

```diff
@@ -142,31 +142,38 @@ def strategy(state: GlobalState, agent_id: str):
     return TradeStrategy(strategy, name=f'back and forth (${percentage})')
 
 
-def invest_all(pool_id: str, assets: list or str = None) -> TradeStrategy:
+def invest_all(pool_id: str, assets: list or str = None, when: int = 0) -> TradeStrategy:
 
     if assets and not isinstance(assets, list):
         assets = [assets]
 
-    def strategy(state: GlobalState, agent_id: str):
+    class Strategy:
+        def __init__(self, _when):
+            self.when = _when
+            self.done = False
 
-        agent: Agent = state.agents[agent_id]
+        def execute(self, state: GlobalState, agent_id: str):
+            if self.done or state.time_step < self.when:
+                return state
+            self.done = True
+            agent: Agent = state.agents[agent_id]
 
-        for asset in assets or agent.holdings.keys():
+            for asset in assets or agent.holdings.keys():
 
-            if asset in state.pools[pool_id].asset_list:
-                state = add_liquidity(
-                    old_state=state,
-                    pool_id=pool_id,
-                    agent_id=agent_id,
-                    quantity=state.agents[agent_id].holdings[asset],
-                    tkn_add=asset
-                )
+                if asset in state.pools[pool_id].asset_list:
+                    state = add_liquidity(
```

Comment:
The reason that invest_all broke was that it uses add_liquidity instead of execute_add_liquidity. Although that was fixed in run.py, it would perhaps be good to switch to using execute_add_liquidity, which I believe brings the invest_all trade strategy more in line with other trade strategies?